doc/rules/control_structure/fully_multiline.rst

```diff
@@ -0,0 +1,94 @@
+========================
+Rule ``fully_multiline``
+========================
+
+Multi-line arrays, arguments list, parameters list and control structures,
+``switch`` cases and ``match`` expressions should have one element by line.
+
+Configuration
+-------------
+
+``elements``
+~~~~~~~~~~~~
+
+Which expression must have one element by line (PHP >= 8.0 for ``parameters``
+and ``match``).
+
+Allowed values: a subset of ``['arguments', 'arrays', 'case', 'control_structures', 'match', 'parameters']``
+
+Default value: ``['arrays']``
+
+Examples
+--------
+
+Example #1
+~~~~~~~~~~
+
+*Default* configuration.
+
+.. code-block:: diff
+
+   --- Original
+   +++ New
+    <?php
+   -array(1,
+   +array(
+   +1,
+        2
+    );
+
+Example #2
+~~~~~~~~~~
+
+With configuration: ``['elements' => ['arguments']]``.
+
+.. code-block:: diff
+
+   --- Original
+   +++ New
+    <?php
+   -foo(1,
+   +foo(
+   +1,
+        2
+    );
+
+Example #3
+~~~~~~~~~~
+
+With configuration: ``['elements' => ['control_structures']]``.
+
+.. code-block:: diff
+
+   --- Original
+   +++ New
+    <?php
+   -if ($a
+   +if (
+   +$a
+        && $b
+    ) {};
+
+Example #4
+~~~~~~~~~~
+
+With configuration: ``['elements' => ['case']]``.
+
+.. code-block:: diff
+
+   --- Original
+   +++ New
+    <?php
+    switch ($foo) {
+   -    case 0: case 1:
+   +    case 0:
+   +case 1:
+            return null;
+        };
+References
```

Comment:
TODO, this rule should be added to the PSR12 with the option `control_structure`.